generators/server/templates/src/main/java/package/config/OAuth2TokenServicesConfiguration.java.ejs

```diff
@@ -0,0 +1,68 @@
+<%#
+ Copyright 2013-2018 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.config;
+
+import org.springframework.boot.autoconfigure.security.oauth2.resource.AuthoritiesExtractor;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.PrincipalExtractor;
+<%_ if (applicationType === 'microservice') { _%>
```

Comment:
The condition should be (applicationType === 'microservice' && cacheProvider !== 'no')

=> See previous version of SecurityConfiguration : CachedUserInfoTokenServices is not used if cacheProvider= 'no'
( However, "no cache" is probably not a good idea in the case ....=)

=> here instead of changing userInfoTokenServices(...) implementation, it is more straightforward to just doesn't generate it (Spring Boot creates the UserInfoTokenService itself in this case)